pom.xml

```diff
@@ -106,6 +135,14 @@
                 <groupId>org.springframework.boot</groupId>
                 <artifactId>spring-boot-maven-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>16</source>
```

Comment:
This project is configured to use Java 16 or Java 21 ? The current configuration contradicts the `<java.version>21</java.version>` setting.